Resolve #721 -- Use Vector2 for postions

lucaheft opened the pull request:
Target now uses a Vector2 for position instead of two floats X and Y

To work LeagueSandbox/LeagueSandbox-Default#82 needs to be merged as well.

Refs #721

Commit:
Resolve #721 -- Use Vector2 for postions

Target now uses a Vector2 for position instead of two floats X and Y

Refs #721

molenzwiebel commented on GameServerLib/Chatbox/Commands/MobsCommand.cs:
```diff
@@ -45,7 +45,7 @@ public override void Execute(Peer peer, bool hasReceivedArguments, string argume
             foreach (var unit in units)
             {
-                 _game.PacketNotifier.NotifyPing(client, unit.Value.X, unit.Value.Y, 0, Pings.PING_DANGER);
+                 _game.PacketNotifier.NotifyPing(client, unit.Value.Position.X, unit.Value.Position.Y, 0, Pings.PING_DANGER);
```
Maybe change `NotifyPing` to also take a `Vector2`?

lucaheft replied:
Yeah i left that as is, because im not so familiar with the networking functionality, also if i remember right Networking is currently worked on and will change.

molenzwiebel commented on GameServerLib/GameObjects/AttackableUnits/AI/Champion.cs:
```diff
@@ -49,7 +49,7 @@ public class Champion : ObjAiBase, IChampion
                         ClientInfo clientInfo,
                         uint netId = 0)
-            : base(game, model, new Stats.Stats(), 30, 0, 0, 1200, netId)
+            : base(game, new Vector2(0, 0), model, new Stats.Stats(), 30, 1200, netId)
```
`Vector2.Zero`

molenzwiebel commented on GameServerLib/GameObjects/Other/Target.cs:
```diff
@@ -6,46 +6,48 @@ namespace LeagueSandbox.GameServer.GameObjects.Other
+        public float GetDistanceToSqr(float x, float y)
         {
-            return new Vector2(X, Y);
+            return (Position.X - x) * (Position.X - x) + (Position.Y - y) * (Position.Y - y);
```
`Vector2` already has a `Distance`/`DistanceSquared` method you can use.

molenzwiebel commented on GameServerLib/GameObjects/AttackableUnits/AI/Minion.cs:
```diff
@@ -26,7 +26,7 @@ public class Minion : ObjAiBase, IMinion
             List<Vector2> mainWaypoints,
             uint netId = 0
-        ) : base(game, "", new Stats.Stats(), 40, 0, 0, 1100, netId)
+        ) : base(game, new Vector2(0, 0), "", new Stats.Stats(), 40, 1100, netId)
```
`Vector2.Zero`

Commit:
Implement requested changes

- use Vector.zero instead of new Vector(0, 0)
- make use of methods Length() and LengthSquared() of Vector2

Commit:
Merge branch 'indev' into TargetPosUpdate

Commit:
Use userId instead of peer for commands

FurkanS1821 commented on GameServerCore/Domain/GameObjects/IChampion.cs:
```diff
@@ -17,6 +17,7 @@ public interface IChampion : IObjAiBase
         void UpdateSkin(int skinNo);
         int GetChampionHash();
+        // Is this needed? We also have StopMovement() in ObjAiBase
         void StopChampionMovement();
```
I doubt that is needed, but i don't think it should be done in this PR. You might want to open an issue for that.

danil179 commented:
There are many conflicts, repo was deleted and it was inactive for a while. Need someone to solve all these conflicts. (will do it myself maybe)

lucaheft commented:
> There are many conflicts, repo was deleted and it was inactive for a while. Need someone to solve all these conflicts. (will do it myself maybe)

Unfortunately im not able to author this any more as im currently writing thesis. Feel free to solve the conflicts. Should be trivial work.

Keep in mind that LeagueSandbox/LeagueSandbox-Default#82 has to be updated as well

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/LeagueSandbox/GameServer?pullRequest=735) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/LeagueSandbox/GameServer?pullRequest=735) before we can accept your contribution.<br/><hr/>**Unknown** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/LeagueSandbox/GameServer?pullRequest=735) it.</sub>

danil179 commented:
Seems like #1036 fixed the Vector2 uses in the code (not completely though). I'm currently closing this PR as it wasn't merged on time. Thank you very much for your contribution and I'm very sorry that it wasn't merged on time.
